opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -0,0 +1,343 @@
+        super().__init__(name, version=version, schema_url=schema_url)
+        self._lock = RLock()
+        self._real_meter: Optional[Meter] = None
+        self._noop_meter = _DefaultMeter(
+            name, version=version, schema_url=schema_url
+        )
+
+    @property
+    def _meter(self) -> Meter:
+        with self._lock:
+            if self._real_meter is not None:
+                return self._real_meter
+
+            if _METER_PROVIDER:
+                self._real_meter = _METER_PROVIDER.get_meter(
+                    self._name,
+                    self._version,
+                )
+                return self._real_meter
+            return self._noop_meter
+
+    def create_counter(self, *args, **kwargs) -> Counter:
+        with self._lock:
+            super().create_counter(*args, **kwargs)
+            return self._meter.create_counter(*args, **kwargs)
+
+    def create_up_down_counter(self, *args, **kwargs) -> UpDownCounter:
+        with self._lock:
+            super().create_up_down_counter(*args, **kwargs)
+            return self._meter.create_up_down_counter(*args, **kwargs)
+
+    def create_observable_counter(self, *args, **kwargs) -> ObservableCounter:
+        with self._lock:
+            super().create_observable_counter(*args, **kwargs)
+            return self._meter.create_observable_counter(*args, **kwargs)
+
+    def create_histogram(self, *args, **kwargs) -> Histogram:
+        with self._lock:
+            super().create_histogram(*args, **kwargs)
+            return self._meter.create_histogram(*args, **kwargs)
+
+    def create_observable_gauge(self, *args, **kwargs) -> ObservableGauge:
+        with self._lock:
+            super().create_observable_gauge(*args, **kwargs)
+            return self._meter.create_observable_gauge(*args, **kwargs)
+
+    def create_observable_up_down_counter(
+        self, *args, **kwargs
+    ) -> ObservableUpDownCounter:
+        with self._lock:
+            super().create_observable_up_down_counter(*args, **kwargs)
+            return self._meter.create_observable_up_down_counter(
+                *args, **kwargs
+            )
+
+
+class _DefaultMeter(Meter):
+    def create_counter(self, name, unit="", description="") -> Counter:
+        with self._lock:
+            super().create_counter(name, unit=unit, description=description)
+            return DefaultCounter(name, unit=unit, description=description)
+
+    def create_up_down_counter(
+        self, name, unit="", description=""
+    ) -> UpDownCounter:
+        with self._lock:
+            super().create_up_down_counter(
+                name, unit=unit, description=description
+            )
+            return DefaultUpDownCounter(
+                name, unit=unit, description=description
+            )
+
+    def create_observable_counter(
+        self, name, callback, unit="", description=""
+    ) -> ObservableCounter:
+        with self._lock:
+            return DefaultObservableCounter(
```

Comment:
Did you mean to not call the super class here?

Reply:
Yes, good catch, fixed :+1: